commit 140761f8a2df10c45eb02041fad5e8237bf57c20
Author: Dan Mick <dmick@redhat.com>
Date:   Tue Jun 19 20:36:08 2018 -0700

    mgr, pybind/mgr: pass inbuf (ceph -i <file>) to modules
    
    Modules may wish to receive bulk data; allow it
    
    Signed-off-by: Dan Mick <dan.mick@redhat.com>

diff --git a/src/mgr/ActivePyModule.cc b/src/mgr/ActivePyModule.cc
index 2f52dd62262..42ad021d02d 100644
--- a/src/mgr/ActivePyModule.cc
+++ b/src/mgr/ActivePyModule.cc
@@ -104,6 +104,7 @@ void ActivePyModule::notify_clog(const LogEntry &log_entry)
 
 int ActivePyModule::handle_command(
   const cmdmap_t &cmdmap,
+  const bufferlist &inbuf,
   std::stringstream *ds,
   std::stringstream *ss)
 {
@@ -122,9 +123,12 @@ int ActivePyModule::handle_command(
   PyFormatter f;
   cmdmap_dump(cmdmap, &f);
   PyObject *py_cmd = f.get();
+  string instr;
+  inbuf.copy(0, inbuf.length(), instr);
 
   auto pResult = PyObject_CallMethod(pClassInstance,
-      const_cast<char*>("handle_command"), const_cast<char*>("(O)"), py_cmd);
+      const_cast<char*>("handle_command"), const_cast<char*>("s#O"),
+      instr.c_str(), instr.length(), py_cmd);
 
   Py_DECREF(py_cmd);
 
diff --git a/src/mgr/ActivePyModule.h b/src/mgr/ActivePyModule.h
index 85d50807a05..ebb1ba182be 100644
--- a/src/mgr/ActivePyModule.h
+++ b/src/mgr/ActivePyModule.h
@@ -54,6 +54,7 @@ public:
 
   int handle_command(
     const cmdmap_t &cmdmap,
+    const bufferlist &inbuf,
     std::stringstream *ds,
     std::stringstream *ss);
 
